config/configmapprovider/file.go

```diff
@@ -43,9 +48,65 @@ func (fmp *fileMapProvider) Retrieve(_ context.Context, _ func(*ChangeEvent)) (R
 		return nil, fmt.Errorf("error loading config file %q: %w", fmp.fileName, err)
 	}
 
+	// Ensure that watches are only created once as Retrieve is called each time that `onChange()` is invoked.
+	if !fmp.watching {
+		watchFile(ctx, fmp.fileName, onChange)
+		fmp.watching = true
+	}
+
 	return &simpleRetrieved{confMap: cp}, nil
```

Comment:
I don't this will as expected by the `Provider` interface. The `Retrieved` that is returned is expected to implement a `Close` function that stops watching and guarantees that `onChange` will not be called after that. See https://github.com/open-telemetry/opentelemetry-collector/blob/96a882a87742e2a921ebabf8e43ad97d8bf0e8c0/config/configmapprovider/provider.go#L76